### PR TITLE
Issue 23 viewcontrollers start using accounts model

### PR DESCRIPTION
**Pull request by shariefyoussef**
I added IBOutlets and IBActions to hand login and new account creation. Created a new UsersRepository to persist and manage new accounts. Updated the Users struct code.
Not done yet with the complete login but I have now the new account creation view along
with some verifications. I will next take the information of the new account and save it into
the new UsersRepository I created so I can compare against when logging in a user.

### CODE REVIEW

**Comment by shariefyoussef** [comment]
I resolved the conflict with the MainStoryboard.

File: GYBITG/NewAccountViewController.swift [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    @IBOutlet weak var firstName: UITextField!
+    @IBOutlet weak var lastName: UITextField!
+    @IBOutlet weak var highSchoolName : UITextField!
+    @IBOutlet weak var clubTeam : UITextField!
+
+    @IBOutlet weak var graduationYear: UIPickerView!
+    
+    @IBOutlet weak var state: UIPickerView!
+    
+    var usersRepository = UsersRepository()
```
Comment (jc-hayes) [inline GYBITG/NewAccountViewController.swift:24]: You may want to check out how Juanita and I instantiate our repos in the AppDelegate file. If you do it directly in the view controller, then every time you leave that view controller you will lose the data.

**Review by jc-hayes — APPROVED** [review]

**Review by Juanitaag1 — APPROVED** [review]

**Review by MeyAyre — APPROVED** [review]

File: GYBITG/NewAccountViewController.swift [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            alert.addAction(UIAlertAction(title: "Cancel", style: UIAlertAction.Style.default, handler: nil))
+            self.present(alert, animated: true, completion: nil)
+        } else if (clubTeam.text == "") {
+            let alert = UIAlertController(title: "Alert", message: "Please enter your club team", preferredStyle: UIAlertController.Style.alert)
+            alert.addAction(UIAlertAction(title: "Cancel", style: UIAlertAction.Style.default, handler: nil))
+            self.present(alert, animated: true, completion: nil)
+        } else {
+            _ = usersRepository.addUser(userToAdd: User(userId: emailAddress.text!, password: password.text!, createDate: Date(), lastLoginDate: Date()))
+            self.dismiss(animated: true, completion: nil)
+        }
```
Comment (Bansenauer-Cascadia) [inline GYBITG/NewAccountViewController.swift:107]: Consider refactoring this repetitive code by extracting the differing portion of each section as a variable, and writing the common portion only once.

File: GYBITG/Repositories/UsersRepository.swift [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+//The protocol for the VideoView
+protocol UsersRepositoryProtocol{
+    //to store all videos
+    func addUser(userToAdd: User) -> String
+    
+}
+
+class UsersRepository: UsersRepositoryProtocol{
+  
+    internal var users = [User]()
```
Comment (Bansenauer-Cascadia) [inline GYBITG/Repositories/UsersRepository.swift:19]: See James' comment about using the AppDelegate to retain the list of users across ViewControllers

**Review by Bansenauer-Cascadia — COMMENTED** [review]
The comments here are not meant to slow down the wrap up of the quarter for you; they could be completed as part of a summer internship. However, the Main.storyboard merge may well be challenging as its not easy integrate changes across conflicting branches of a storyboard. 

Be sure to pull develop and merge it into you issue-23-viewcontrollers-start-using-accounts-model branch first to see whether you can get the resulting conflicts successfully resolved. Run and View your result before pushing the branch back up to GitHub.

**Pull request merged**